sdcm/utils/sstable/sstable_utils.py

```diff
@@ -131,39 +131,30 @@ def check_that_sstables_are_encrypted(self, sstables=None,  # pylint: disable=to
             f" Success part: '{encryption_success_part}'. Expected bool value: '{expected_bool_value}'."
             f" Encryption results: {encryption_results}")
 
-    def count_sstable_tombstones(self, sstable: str) -> int:
+    def _count_ttl_expired_tombstones(self, sstable: str) -> int:
         """
-        Counts the number of tombstones in a given SSTable.
+        Counts the number of ttl-expired tombstones in a given SSTable.
 
         :param sstable: The SSTable file path.
-        :return: The number of tombstones in the SSTable, or 0 if SSTable doesn't exist.
+        :return: The number of ttl-expired tombstones in the SSTable, or 0 if SSTable doesn't exist.
         """
         if not self._run_sstabledump(sstable=sstable):  # Check if SSTable exists and was dumped
             self.log.debug("Skipping tombstone count as SSTable %s does not exist or dump failed.", sstable)
             return 0
 
-        # Fetch the dumped JSON content
         result = self.db_node.remoter.run(
-            f'sudo cat {self.REMOTE_SSTABLEDUMP_PATH}', verbose=False, ignore_status=False)
-
-        if not result.ok:
-            self.log.debug("Failed to retrieve SSTable dump data for %s: (%s, %s)", sstable, result.stdout,
-                           result.stderr)
-            return 0
-
-        try:
-            dump_data = json.loads(result.stdout)
-            num_tombstones = sum(
-                1 for partition in dump_data.get("sstables", {}).get("anonymous", [])
-                if "tombstone" in partition or partition.get("expired") is True
-            )
-
-            self.log.debug("Found %s tombstones in SSTable %s", num_tombstones, sstable)
+            f"sudo awk -F 'deletion_time' '{{print NF-1}}' {self.REMOTE_SSTABLEDUMP_PATH}",
+            verbose=True,
+            ignore_status=False
+        )
+
+        if result.ok:
+            num_tombstones = int(result.stdout.strip())
+            self.log.debug("Found %d TTL-expired tombstones in SSTable %s", num_tombstones, sstable)
             return num_tombstones
-
-        except json.JSONDecodeError as e:
-            self.log.error("Failed to parse SSTable dump JSON for %s: %s", sstable, str(e))
-            raise
+        else:
+            self.log.debug("Failed to count TTL-expired tombstones in %s dump: %s", sstable, result.stderr)
+            return 0
```

Comment:
If we fail to count them, shouldnt we throw an exception? By returning zero the test might pass because we didnt count them

Reply:
to be on the safe side, i'll raise an exception for that.